sw/device/silicon_creator/rom_ext/doc/rescue.md

```diff
@@ -0,0 +1,143 @@
+# ROM_EXT Rescue Protocol
+
+## Introduction
+
+The ROM_EXT rescue protocol permits the chip owner to perform device diagnostic and management actions when there is no functional application firmware.
+
+These actions include:
+- Flashing a new application image.
+- Interatcting with Boot Services.
+- Requesting the Boot Log data.
+- Performing Ownership Transfer.
+
+Currently, the rescue protocol uses the console UART as the main communication channel.
+In the future, rescue operations may be available via additional interfaces.
+
+## Basic Operation
+
+In order to interact with the rescue protocol, the user needs to signal to the ROM_EXT that rescue a rescue operation is desired.
+
+Once signalled, the ROM_EXT will acknowledge entry into rescue mode and the user can then specify a specific rescue action and either upload or download any data block associated with that action.
+
+Rescue mode is exited via either an in-band or out-of-band reset signal.
+
+## Serial Rescue Protocol
+
+### Requesting Rescue
+
+In order to signal to the ROM_EXT that a rescue operation is desired, the user should assert the serial break condition for at least 350us while the ROM_EXT boots.
+When the ROM_EXT detects the serial break condition, it will respond with an acknowledgement:
+
+```
+rescue: remember to clear break
+```
+
+Upon entry into rescue mode, the user should clear the serial break condition to allow further interaction with rescue mode.
+
+### Rescue Operations
+
+By default, rescue will enter firmware rescue mode and will prompt the user to upload a new firmware image via the Xmodem-CRC protocol.
+Alternate modes are requested by sending the mode's 4-byte code followed by a newline character.
+
+The following sections detail each of the supported alternate modes.
+
+#### Firmware Rescue (`RESU`)
+
+The firmware rescue mode may be requested with the 4-byte code `RESU`.
+The ROM_EXT will acknowledge entry of this mode with the following message:
+
+```
+mode: RESU
+ok: send firmware via xmodem-crc
+```
+
+The ROM_EXT will then will prompt for the transfer to start by sending the Xmodem-CRC start character (which is the ASCII character `C`).
+
+#### Request Boot Log Data (`BLOG`)
```

Comment:
Should the ownership blob have an policy in the rescue section to allow-list which operations are available in rescue mode? This is mainly so that we can give Owners with certification requirements the ability to disable commands that may give an advantage to a potential attacker.

Reply:
I think yes.  We talked about the ownership blob defining which ranges could be flashed with rescue protocol, and I also think we want the blob to define that other data and info ranges of flash will be erased on rescue (or  will be kept as-is, and cannot be overwritten by rescue).  The owner being able to control other features of the rescue logic, and being able to disable it altogether (such that a grounded RX pin does not even cause 0.1s delay) sounds reasonable.

Reply:
Yes, I agree: the ownership block should specify the enabled/disabled features of the rescue protocol.